src/utils/deep-linking.utils.ts

```diff
@@ -0,0 +1,45 @@
+import { Serializer } from '@airgap/beacon-sdk';
+import { useEffect } from 'react';
+import { Linking } from 'react-native';
+import { URL } from 'react-native-url-polyfill';
+
+import { BeaconHandler, isBeaconMessage } from '../beacon/beacon-handler';
+import { isDefined } from './is-defined';
+
+export const tezosDeepLinkHandler = async (url: string) => {
+  try {
+    const searchParams = new URL(url).searchParams;
+    const type = searchParams.get('type');
+    const data = searchParams.get('data');
+
+    if (type === 'tzip10' && isDefined(data)) {
+      const json = await new Serializer().deserialize(data);
+      if (isBeaconMessage(json)) {
+        await BeaconHandler.addPeer(json);
+      }
+    }
+  } catch {}
+};
+
+export const deepLinkHandler = async (url: string) => {
+  const href = new URL(url).href;
+  switch (href) {
+    case 'temple://':
+      break;
+    case 'tezos://':
+      await tezosDeepLinkHandler(url);
+      break;
```

Comment:
`temple` && `tezos` deeplinks should be handled in the same way, so there is no need in `switch`

Reply:
there will be if/else condition anyway, i just classified different variants of possible handlings